sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -223,6 +227,16 @@ public void setString(int ordinal, String value) {
   }
 
   @Override
+  public void setDate(int ordinal, Date value) {
+    throw new UnsupportedOperationException();
```

Comment:
Date is a fixed-width type that's stored in the primitive values section, so I think that we should be able to support in-place updates to it.  I think that we can just move the `setInt` call from `DateUnsafeColumnWriter` to here, then update that method to call `target.setDate` here.